src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -1807,6 +1877,160 @@ public override TypeKind TypeKind
             }
         }
 
+        private bool IsExtensionType([NotNullWhen(true)] out MethodDefinitionHandle? foundMarkerMethod)
```

Comment:
>private bool IsExtensionType([NotNullWhen(true)] out MethodDefinitionHandle? foundMarkerMethod) [](http://example.com/codeflow?start=8&length=95)

Not sure why do we need to use nullable type. I think there is a notion of Nil handle, which indicates the lack of value. Also, Consider simply returning the ```MethodDefinitionHandle```, this is going to eliminate the need to coordinate both values and assume that one is not null based on other boolean value. The method could be called "TryGetMarkerMethod" then.